src/simtools/simtel/simtel_io_event_reader.py

```diff
@@ -0,0 +1,262 @@
+            telescope_list_array = f.root.data.trigger_telescope_list_list
+
+            triggered_event_data.trigger_telescope_list_list = []
+            for index in telescope_indices:
+                if index < telescope_list_array.nrows:
+                    triggered_event_data.trigger_telescope_list_list.append(
+                        telescope_list_array[index]
+                    )
+                else:
+                    self._logger.warning(f"Invalid telescope list index: {index}")
+                    triggered_event_data.trigger_telescope_list_list.append(
+                        np.array([], dtype=np.int16)
+                    )
+
+        triggered_shower, triggered_data = self._reduce_to_triggered_events(
+            event_data, triggered_event_data
+        )
+        return event_data, triggered_shower, triggered_data
+
+    def _reduce_to_triggered_events(self, event_data, triggered_data):
+        """
+        Reduce event data to triggered events only. Apply filter on telescope list if specified.
+
+        Parameters
+        ----------
+        event_data : ShowerEventData
+            Event data.
+        triggered_data : TriggeredEventData
+            Triggered event data.
+
+        Returns
+        -------
+        ShowerEventData, TriggeredEventData
+            Filtered event data and triggered event data
+        """
+        filtered_shower_ids, triggered_indices = self._get_mask_triggered_telescopes(
+            self.telescope_list,
+            triggered_data.triggered_id,
+            triggered_data.trigger_telescope_list_list,
+        )
+        filtered_event_data = ShowerEventData(
+            x_core=event_data.x_core[filtered_shower_ids],
+            y_core=event_data.y_core[filtered_shower_ids],
+            simulated_energy=event_data.simulated_energy[filtered_shower_ids],
+            shower_azimuth=event_data.shower_azimuth[filtered_shower_ids],
+            shower_altitude=event_data.shower_altitude[filtered_shower_ids],
+        )
+
+        filtered_telescope_list = [
+            triggered_data.trigger_telescope_list_list[i] for i in triggered_indices
+        ]
+
+        filtered_triggered_data = TriggeredEventData(
+            array_azimuths=triggered_data.array_azimuths[triggered_indices],
+            array_altitudes=triggered_data.array_altitudes[triggered_indices],
+            trigger_telescope_list_list=filtered_telescope_list,
+        )
+        self._logger.info(
+            f"Events reduced to triggered events: {len(filtered_event_data.simulated_energy)}"
+        )
+        return filtered_event_data, filtered_triggered_data
+
+    def _get_mask_triggered_telescopes(
+        self, telescope_list, triggered_id, trigger_telescope_list_list
+    ):
+        """
+        Return indices of events that triggered the specified telescopes.
+
+        Parameters
+        ----------
+        telescope_list : list
```

Comment:
other parmeters missing

Reply:
Are you sure? Which one is missing?

https://github.com/gammasim/simtools/blob/be9495e2edba0b5986226fe16ddd06d77475096d/src/simtools/simtel/simtel_io_event_reader.py#L151